presidio-anonymizer/app.py

```diff
@@ -22,19 +25,38 @@ def __init__(self):
         def anonymize():
             content = request.get_json()
             if not content:
-                return "Invalid request json", 400
+                return ErrorResponse("Invalid request json"), 400
             try:
-                data = AnonymizerRequest(content)
-                text = AnonymizerEngine().anonymize(data)
+                engine = AnonymizerEngine()
+                data = AnonymizerRequest(content, engine.builtin_anonymizers)
+                text = engine.anonymize(data)
             except InvalidParamException as e:
                 self.logger.warning(
                     f"failed to anonymize text with validation error: {e.err_msg}")
                 return e.err_msg, 422
             except Exception as e:
                 self.logger.error(f"failed to anonymize text with error: {e}")
-                return "Internal server error", 500
+                return ErrorResponse("Internal server error"), 500
             return text
 
+        @self.app.route("/health")
+        def health() -> str:
+            """Return basic health probe result.  get ok + 200."""
+            return "ok"
+
+        @self.app.route("/anonymizers", methods=["GET"])
+        def anonymizers() -> Tuple[str, int]:
+            """Return a list of supported anonymizers."""
+            try:
+                return json.dumps(AnonymizerEngine().anonymizers()), 200
```

Comment:
we can utilize the singleton engine here as well

Reply:
Please see my thread about singletons in the channel